GitExtUtils/GitArgumentBuilder.cs

```diff
@@ -171,22 +98,40 @@ public void Add(GitConfigItem configItem)
             }
         }
 
-        /// <inheritdoc />
-        public override string ToString()
+        /// <summary>
+        /// Gets the length of the expected command.
+        /// </summary>
+        /// <returns>THe length of the expected command.</returns>
+        public override int GetLength()
         {
-            var arguments = base.ToString();
             var gitArgsLength = _gitArgs.Length;
             if (gitArgsLength == 0)
             {
                 gitArgsLength = -1; // prevent extra capacity when length is 0
             }
 
             // 7 = "-c " + "=" + " " + 2 (for possible quotes of value)
-            var capacity = _configItems.Sum(i => i.Key.Length + i.Value.Length + 7) + _command.Length + 1 + arguments.Length;
```

Comment:
@RussKie  @mstv  I don't think adding `2` is correct. If you think of `GetLength` as the exact length of the command, we should not `+2` in case config item does not `NeedEscaping`. So `Length` is operating like `Capacity`, and we can easily get confused when using `GetLength`, and just received `Capacity` instead.

Reply:
> So `Length` is operating like `Capacity`, and we can easily get confused when using `GetLength`, and just received `Capacity` instead.

This is a fair remark. However in the context of the change you're proposing for breaking up a long command - does it really matter? A capacity is the top boundary (i.e. a worse case scenario). 
The reason we precompute the capacity (i.e. the max length of the command) is to ensure we allocate only once.

Reply:
It'll work fine for long command case, but we must also change the boundary test cases to adapt this logic. That and the logic itself would be confusing. 
My point is we could add a extra calculation, so `Length` would be exact length. For example:
```csharp
bool isQuoted = configItem.Value.StartsWith("\"", StringComparison.Ordinal) && configItem.Value.EndsWith("\"", StringComparison.Ordinal);
length += configItem.Key.Length + configItem.Value.Length + (isQuoted ? 7 : 5);
```